### DISCUSSION TITLE
Search documents by content not only title

### DISCUSSION THREAD

**Discussion by nrygh** [jonaswinkler/paperless-ng#614] [Archived FRs]
In version 1.2.0 I'm able to do a fulltext (content) search in the top (green bar).
Output is a list of all documents found, with probability, but features from documents section filtering are missing (select, add tag, bulk download...)

What I would like to have is the options of the filtering under documents.

-select found documents
-add tag, correspondant,.. to selection
-or download all selected files as .zip
-change views

I don't know what is best?

a) add the features from the document filter to the search in the top
b) add context search to the document section

Plus:
It would be nice to have a option for the document retagger to specify which tag to work on (not use all tags).
#mamage.py document_retagger -T 'tagstring'

Regards

**Comment by jonaswinkler** [comment]
See #357, and the linked comment in there. This is tough and probably won't happen.

**Reply by nrygh** [reply]
In old paperless it was possible. If you have access to old paperless you could simply search, mark and then apply a action.
Maybe you could implement a select box in the document section to decide whether to search in title and/or content?

**Reply by jonaswinkler** [reply]
![image](https://user-images.githubusercontent.com/17569239/109012264-1d22ac80-76b2-11eb-8c99-ad3911bf98b2.png)

Should I change this field to search both the title and content, instead of just the title?

**Reply by nrygh** [reply]
yes, or to have the option to decide to search title and/or content (with a small checkbox or the possibility to decide in the configuration section what to search, or a dropdown menu instead of the checkbox)?

**Reply by shamoon** [reply]
IIRC it wasn’t trivial to change that field to include searching content because different search mechanisms / API would need changing?

Obviously your call JW but there are a few other backlogged FRs with multiple upvotes (and lingering PRs)...

**Reply by jonaswinkler** [reply]
* It's relatively simple to adjust that to search the content as well. This will just be a very simple "does the content include the string" search.

* It's not easy to use the actual full text searching capabilities here. These provide ranked results (relevant documents show up first) and highlighting, which is something this filter cannot provide.

> but there are a few other backlogged FRs with multiple upvotes (and lingering PRs)...

I know.

**Reply by shamoon** [reply]
Ah that makes sense. Well yea sounds useful. IMHO I’d say probably don’t bother with a checkbox or drop down, just change the placeholder to be “Title / Content” or whatever. Keep UI simpler. Maybe you were already thinking that :)
